### ISSUE TITLE
failed at polish step

### ISSUE THREAD

**Issue by yuhuang-genomics** [aidenlab/3d-dna#119]
Dear authors,

I'm running the run-asm-pipeline.sh with default parameters and failed at the first(polish) step (probably **visualize/run-asm-visualizer.sh**). Log file says ":( Assembly file does not match cprops file. Exiting!"* multiple lines.  

**merged_nodups.txt** looks fine except it is super large (**size: 67Gb**)
0 ctg9_np1212 13629867 32020 16 ctg9_np1212 13630208 32020 60 25S125M CCGTCTGCTTCTCseq... 60 150M AAGGTTTTGAGGGTseq... V350011968L1C002R0191101361/1 V350011968L1C002R0191101361/2
...

**genome.fasta header:**
>ctg2051_np1212 12374
>ctg917_np1212 53277
>ctg1357_np1212 52513
... 

Iterations have completed and I got the following files:
**genome.0.cprops (size:34K):**
ctg6_np1212 1 25479283
ctg2_np1212 2 24769909
ctg5_np1212 3 22795029
...

**genome.0.asm (size: 6.3K**; It is a binary file, which confuses me a lot. The h.*txt files generated in iterations are normal readable files. How does the final step "cp "h.scaffolds.original.notation.step.""$STEP"".txt" "$basenamefile"".asm"" generate a binary file?)

**genome.0.assembly (size: 41k)**
>ctg6_np1212 1 25479283
>ctg2_np1212 2 24769909
>ctg5_np1212 3 22795029
...

**genome.0_asm.superscaf_track.txt (size: 5k)**
chr1	sx1	sx2	chr2	sy1	sy2	color	Superscaffold_ID	x1	x2	y1	y2
assembly	0	0	assembly	0	0	0,0,255	1	0	0	0	0
assembly	0	40251	assembly	0	40251	0,0,255	2	0	40251	0	40251
assembly	40251	86719	assembly	40251	86719	0,0,255	3	40251	86719	40251	86719
...

**genome.0_asm.scaffold_track.txt (EMPTY)**  There might be something wrong in lift/lift-input-annotations-to-asm-annotations.awk. The  binary asm file is used as an input.

No .hic outputs.

I have reran the pipeline for three times and it always stopped here. BYW, do you have an example with tmp files so that I could check myself which file is incorrect.

Thanks in advance.
Yu

**Comment by dudcha** [comment]
Hello Yu,

You say it is first(polish) step which is confusing to me. The first step is not polish. So, are you observing an error during the first step or during the polish step? The fact that  genome.0_asm.scaffold_track.txt is empty suggests you have an issue with the first scaffolding step.

genome.0.asm should not be, and is, not a binary file. Why are you saying it is a binary file? The genome.0.assembly is clearly readable, and .asm is just part of that file.

Can you share the stdout and stderr?

Best,
Olga

**Comment by yuhuang-genomics** [comment]
Dear Olga,

Thanks very much for your kind reply. I really appreciate it. 
Sorry for the mistake, it's the first step "ITERATIVE SCAFFOLDING/MISJOIN CORRECTION" that reports the error. 
Yes, it is really odd that genome.0.asm becomes a binary. 
less genome.0.asm
"genome.0.asm" may be a binary file.  See it anyway?

Files are uploaded
[upload.zip](https://github.com/aidenlab/3d-dna/files/6545459/upload.zip)

Thanks,
Yu

**Comment by dudcha** [comment]
what happens if you open it anyway? -Olga

**Comment by yuhuang-genomics** [comment]
The complete .asm file is included in the uploaded zip.
And it looks like:
@^@^@^@^@^@^@^@^@^@^@^@^@1035
948
829
645
598
470
1082
1149
1216
1180
1267
909
524

**Comment by yuhuang-genomics** [comment]
Dear Olga,

I seem to figure out what the problem is. My genome.fasta sequences contain both lowercase and uppercase letters (polished by Nextpolish), Juicer creates the merge_nodups.txt by ignoring the lowercase seqs, so that the two seq columns have shorter, incomplete seqs. Now I'm rerunning 3d_DNA with the new genome.fasta (all uppercase letters) and merge_nodups, and the first round of iteration outputs the correct ams and tracks. Please keep the issue open in case new problems arise.

Thanks,
Yu

**Comment by dudcha** [comment]
Hi Yu,

Thanks for reporting, but I do not see how lowercase letters in the reference (if by lowercase you mean soft-masking) could be the problem. BWA used internally by Juicer does not ignore them for alignment purposes as far as I know there is no special treatment. Unless these letters are somehow not just lowercase but also have some encoding problem.

Thanks,
Olga

**Comment by yuhuang-genomics** [comment]
Dear Olga,

Finally! Yes, you're right. It's not the problem of a mixed lowercase and uppercase letters. "cp" is where the error occured.
I firstly modified the script to keep "h.*.txt" files and they all look good. Then error happened at the command "cp "h.scaffolds.original.notation.step.""$STEP"".txt"  "$basenamefile"".asm"" as I reported above.  So I changed the "cp" command to "cat h.file >file.asm" in run-liger-scaffolder.sh (and all other scripts with "cp"s). Finally, the pipeline runs and outputs the FINAL files. 

Thanks for your help.
Yu
